source/js/components/search-form.js

```diff
@@ -1,98 +1,106 @@
 import fetchUtils from "@degjs/fetch-utils";
+import formMapper from "@degjs/form-mapper";
 
 const searchForm = () => {
 
 
     const formEl = document.querySelector('.js-search-form');
-    const layoutEl = document.querySelector('.js-general-content')
+    const layoutEl = document.querySelector('.js-general-content');
+    const searchInputEl = formEl.querySelector('.js-search');
+    const searchButtonEl = formEl.querySelector('.js-search-btn');
 
     const url = formEl.dataset.api;
 
     function init() {
-        formEl.addEventListener('submit', e => {
-            e.preventDefault();
-            getData();
-        });
+        formEl.addEventListener('submit', searchSubmit);
     }
 
+    function searchSubmit(e) {
+        e.preventDefault();
+        getData();
+    }
 
     function getData() {
-        const formEls = formEl.querySelectorAll('.js-value');
+        preloader(true);
         fetchUtils.getJSON(
             url,
             {
-                body: JSON.stringify(formEls),
+                body: JSON.stringify(formMapper.getValues(searchInputEl)),
                 method: 'POST',
                 headers: {
                     'Content-Type': 'application/json'
                 }
-            },
-            {
-                timeout: 10000
             }
         )
         .then((data) => {
+            preloader(false);
             renderResults(data.results);
         })
         .catch(error => {
+            preloader(false);
             alert(error);
         });
     }
 
+    function preloader(isSearching) {
```

Comment:
[fix] Good addition of this. I think the name could be improved (something like `showLoading`, maybe?), and you probably also want to lock down the search input box. But good overall.

Reply:
Fixed, added fieldset to disable the entire form.